### DISCUSSION TITLE
Auto-increment Document Identifier (id)

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#371] [Feedback & Feature Proposal]
Hey 👋

We are thinking of adding a feature to automatically generate a document identifier by auto-incrementation.

Would this feature be useful for you? In what case would you use it?

We think that it could help the users who would have friction today because the identifier is mandatory for MeiliSearch while this field is not relevant for them.

**Comment by curquiza** [comment]
As a user (not as a Meili employee), I would love to use this feature in the case I only put "scrapped" data to Meilisearch, like scraping the documentation. I don't need an identifier per document, I just need to push the data. Every time I want to update the data, I just delete the index and create a new one: **I don't need to update them individually.**

Every time I do this, I forgot Meilisearch needs a primary key and I got the following error

```json
"error": {
                "message": "The primary key inference process failed because the engine did not find any fields containing `id` substring in their name. If your document identifier does not contain any `id` substring, you can set the primary key of the index.",
                "code": "primary_key_inference_failed",
                "type": "invalid_request",
                "link": "https://docs.meilisearch.com/errors#primary_key_inference_failed"
            }
```

I can easily fix the error (maybe I say "easily" because I'm a Meili employee?), but it's still a frustration when using Meilisearch.

For me, this feature could be useful for people who don't need to update their document individually, like me.
